### PR TITLE
pull request from new-widgets

### PR DESCRIPTION
**Pull request by bfhub**
Implemented JIRA B1-138

New widgets: Stoplight, simple text-message, and data-table.

### CODE REVIEW

File: src/cljs/vanilla/widgets/table_widget.cljs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+            [vanilla.widgets.basic-widget :as basic]))
+
+
+(defn make-widget [name data options]
+
+  ;(prn ":grid make-widget" name
+  ;     " //// data " data
+  ;     " //// (options) " options)
+
+  ( let [header (get (get data :data) :meta-data)
```
Comment (cawasser) [inline src/cljs/vanilla/widgets/table_widget.cljs]: more idiomatic clojure is:

`(get-in data [:data :meta-data])`

File: src/cljs/vanilla/widgets/table_widget.cljs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+         body (get (get data :data) :series)]
```
Comment (cawasser) [inline src/cljs/vanilla/widgets/table_widget.cljs]: and `(get-in data [:data :series])`

File: src/cljs/vanilla/widgets/stoplight_widget.cljs [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+       ;(.log js/console (str "stoplight-row> " k "-" v))
+
+       (stoplight k v)))])
+
+
+
+(defn make-widget [name data options]
+
+  (prn ":stoplight make-widget" name
```
Comment (cawasser) [inline src/cljs/vanilla/widgets/stoplight_widget.cljs]: comment these out (prn ...) blocks now that things are working, but leave the commented code so we can turn it back on in the future

File: src/cljs/vanilla/widgets/table_widget.cljs [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+        [:tbody
+         (for [b body]
+           [:tr
+             (for [[k v] b]
+               (do
+                 ^{:key (str v)}[:td (str v)]))])]]]]))
+
+
+
```
Comment (cawasser) [inline src/cljs/vanilla/widgets/table_widget.cljs]: pull out these blanks just to make things tidy

File: src/cljs/vanilla/widgets/stoplight_widget.cljs [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    [:button.button {:style    {:background-color c
+                                :width            "20%"
+                                :border-spacing   "5px"}
+                     :on-click #(.log js/console (str "clicked " id))}
+     id]))
+
+
+(defn- stoplight-row [data]
+
+  ;(.log js/console (str "stoplight-row " data))
```
Comment (cawasser) [inline src/cljs/vanilla/widgets/stoplight_widget.cljs]: convert this "old style" to

`(prn "stoplight-row " data)`

style connect blocks

File: src/cljs/vanilla/widgets/core.cljs [hunk 6]
```diff
@@ -3,16 +3,19 @@
...
             [vanilla.widgets.make-chart :as mc]))
 
-(def widget-store (atom {}))
+;(def widget-store (atom {}))
 
 ; TODO - is having a widget registry even worth doing?
 
-(defn register-widget [name w]
-  (swap! widget-store assoc name w))
+;(defn register-widget [name w]
```
Comment (cawasser) [inline src/cljs/vanilla/widgets/core.cljs]: remove all this "dead code"

File: src/cljs/vanilla/widgets/core.cljs [hunk 7]
```diff
@@ -3,16 +3,19 @@
             [vanilla.widgets.basic-widget :as basic]
             [vanilla.widgets.make-chart :as mc]))
 
-(def widget-store (atom {}))
+;(def widget-store (atom {}))
```
Comment (cawasser) [inline src/cljs/vanilla/widgets/core.cljs]: remove this dead code

File: src/cljs/vanilla/widgets/core.cljs (hunk 6 above)
Comment (cawasser) [inline src/cljs/vanilla/widgets/core.cljs]: remove this dead code

File: src/cljs/vanilla/widgets/core.cljs [hunk 8]
```diff
@@ -3,16 +3,19 @@
...
 
-(defn register-widget [name w]
-  (swap! widget-store assoc name w))
+;(defn register-widget [name w]
+;  (swap! widget-store assoc name w))
 
 ;(prn "register-widget " type
 ;  " //// w " w))
 
+;(defn build-simple [type data options]
```
Comment (cawasser) [inline src/cljs/vanilla/widgets/core.cljs]: remove this dead code

**Review by austinpaine — APPROVED** [review]
Agree with chris on general cleanup, looks great!

**Comment by cawasser** [comment]
> Implemented JIRA B1-138

> 
> New widgets: Stoplight, simple text-message, and data-table.

File: src/cljs/vanilla/widgets/core.cljs (hunk 8 above)
Comment (bfhub) [reply src/cljs/vanilla/widgets/core.cljs]: addressed all your comments, and changes are in repo, commit:
194b90df92d28746cc45764451ff5dfa87ea6ec8

**Review by gpierce3 — APPROVED** [review]
No additional comments.

**Pull request merged**